packages/babel-types/src/definitions/core.js

```diff
@@ -538,10 +539,12 @@ defineType("Program", {
   // Note: We explicitly leave 'interpreter' out here because it is
   // conceptually comment-like, and Babel does not traverse comments either.
   visitor: ["directives", "body"],
-  builder: ["body", "directives", "sourceType", "interpreter"],
+  builder: ["body", "directives", "sourceType", "interpreter", "sourceFile"],
```

Comment:
I don't think anyone knows this field is even there and I'm not even sure off the top of my head what it is for, so I'd rather leave this out.